### PR TITLE
add epoch timestamp in IM Event

### PR DESCRIPTION
**Pull request by yunhanw-google**
#### Problem
--For IM Events, it needs support either system timestamp or utc timestamp by spec, platform would pick up either one kind timestamp for IM event logging. which means some platfrom would  not have utc event logging. Before this PR, CHIP_CONFIG_EVENT_LOGGING_UTC_TIMESTAMPS flag is useless. utc im timestamp is disabled by all example application currently,  CHIP_CONFIG_EVENT_LOGGING_UTC_TIMESTAMPS = 0 is at default configuration
this PR pick up freertos platform to enable this utc timestamp feature for IM event for reference where it has #if CHIP_CONFIG_EVENT_LOGGING_UTC_TIMESTAMPS & CHIP_SYSTEM_CONFIG_PLATFORM_PROVIDES_TIME set, for other platform, if they wanna use utc timestamp for IM event, they can refer the freertos implementation
--Remove systemTimestamp's system prefix to suite for both usage for epoch and system.
--Create abstract API GetClock_RealTime, GetClock_RealTimeMS, SetClock_RealTime IN system/Clock.h, implemented it in freertos platform
#### Change overview
See above

#### Testing
Compilation and presubmit tests.

### CODE REVIEW

File: src/system/SystemClock.h [hunk 1]
```diff
@@ -184,6 +184,10 @@ class ClockBase
      * @returns             Elapsed time in milliseconds since an arbitrary, platform-defined epoch.
      */
     virtual Milliseconds64 GetMonotonicMilliseconds64() = 0;
+
+    virtual CHIP_ERROR GetClock_RealTime(uint64_t & curTime)   = 0;
```
Comment (kpschoedel) [inline src/system/SystemClock.h:188]: Can these be made type-safe? Either way, please document what these values are.

**Comment by kpschoedel** [comment]
13K seems pretty bad… Is that avoidable, or inherently pulling in some new large library code?

> | platform   | target           | config   | section     |   2e85d48c |   3af0bd5c |   change |   % change |
>|:-----------|:-----------------|:---------|:------------|-----------:|-----------:|---------:|-----------:|
>| esp32      | all-clusters-app | c3devkit | (read only) |     834242 |     847272 |    13030 |        1.6 |

File: src/system/SystemClock.h (hunk 1 above)
Comment (yunhanw-google) [reply src/system/SystemClock.h]: just update these value.

Comment (yunhanw-google) [reply src/system/SystemClock.h]: I think i can make it type-safe via applying Clock::Milliseconds64

**Review by woody-apple — CHANGES_REQUESTED** [review]
@yunhanw-google It looks like we're disabling UTC_TIMESTAMPS for every platform (unless I read these diffs wrong) Should we remove that code too?


Fast tracking, given there's been enough time for review.

**Comment by yunhanw-google** [comment]
> @yunhanw-google It looks like we're disabling UTC_TIMESTAMPS for every platform (unless I read these diffs wrong) Should we remove that code too?
@woody-apple  thanks for questioning.
Currently no code is using CHIP_CONFIG_EVENT_LOGGING_UTC_TIMESTAMPS if you search code, it is basically empty flag for nothing, in other words, there is no utc/epoch timestamp in Matter code, This PR is firstly adding the UTC/Epoch timestamp and abstract API for utc timestamp in system/clock, for example, I added the minimal support for freertos, platform then for better cleanup, I disable this flag in other platform explicitly in case some platfom needs it in the future and they can enable it.

**Comment by woody-apple** [comment]
> @yunhanw-google It looks like we're disabling UTC_TIMESTAMPS for every platform (unless I read these diffs wrong) Should we remove that code too?
> @woody-apple thanks for questioning.
> Currently no code is using CHIP_CONFIG_EVENT_LOGGING_UTC_TIMESTAMPS if you search code, it is basically empty flag for nothing, in other words, there is no utc/epoch timestamp support in Matter code. This PR is not disabling every platform for utc, and the freertos platform still have it, this PR firstly adds the UTC/Epoch timestamp and abstract API for utc timestamp in system/clock, for example, I added the minimal support for freertos platform, then for better cleanup, I disable this flag in other platforms explicitly since those don't have any utc code support, in case these platform needs it in the future and they can enable it. In IM eventManagement, it would use utc time when CHIP_CONFIG_EVENT_LOGGING_UTC_TIMESTAMPS and CHIP_SYSTEM_CONFIG_PLATFORM_PROVIDES_TIME are set.

Thanks! My main concern here is that this PR claims it's adding timestamp support, but also removes other UTC timestamp support. 

• If we don't need UTC timestamp support, let's remove it entirely. If some platforms require it, how do we make this more abstract?
• If nobody needs it, let's remove it entirely.

If we're doing cleanup like this (which is great!) let's do this cleanup completely.

**Comment by yunhanw-google** [comment]
> > @yunhanw-google It looks like we're disabling UTC_TIMESTAMPS for every platform (unless I read these diffs wrong) Should we remove that code too?
> > @woody-apple thanks for questioning.
> > Currently no code is using CHIP_CONFIG_EVENT_LOGGING_UTC_TIMESTAMPS if you search code, it is basically empty flag for nothing, in other words, there is no utc/epoch timestamp support in Matter code. This PR is not disabling every platform for utc, and the freertos platform still have it, this PR firstly adds the UTC/Epoch timestamp and abstract API for utc timestamp in system/clock, for example, I added the minimal support for freertos platform, then for better cleanup, I disable this flag in other platforms explicitly since those don't have any utc code support, in case these platform needs it in the future and they can enable it. In IM eventManagement, it would use utc time when CHIP_CONFIG_EVENT_LOGGING_UTC_TIMESTAMPS and CHIP_SYSTEM_CONFIG_PLATFORM_PROVIDES_TIME are set.
> 
> Thanks! My main concern here is that this PR claims it's adding timestamp (epoch) support, but also removes other UTC timestamp support.
> 
> • If we don't need UTC timestamp support, let's remove it entirely. If some platforms require it, how do we make this more abstract? • If nobody needs it, let's remove it entirely.
> 
> If we're doing cleanup like this (which is great!) let's do this cleanup completely.

Thanks! Just remove those CHIP_CONFIG_EVENT_LOGGING_UTC_TIMESTAMPS for platform which currently don't use it.

**Review by yufengwangca — APPROVED** [review]

**Comment by yunhanw-google** [comment]
Per discussion with @woody-apple , would file several tickets on epoch timestamp for different platform

**Review by robszewczyk — APPROVED** [review]

**Review by woody-apple — APPROVED** [review]

**Pull request merged**